shared-libs/couch-request/test/couch-request.js

```diff
@@ -0,0 +1,186 @@
+const couch_request = require('../src/couch-request');
+const chai = require('chai').use(require('chai-as-promised'));
+const request = require('request-promise-native');
+const sinon = require('sinon');
+
+class Unicorn { }
+const notPlainObjects = [[1, 2, 3], new Unicorn(), new Map([[1, 1], [2, 2], [3, 3]]), new Set([1, 2, 3]),
+  () => { }, true, 'foo', null, 1, NaN, Infinity, /foo/, new Date(), new Error(), new Int8Array(), new Float32Array(),
+  new Float64Array(), new Uint8Array(), new Uint8ClampedArray(), new Uint16Array(), new Uint32Array(),
+  new ArrayBuffer(), new WeakMap(), new WeakSet()];
```

Comment:
If you use a library like lodash you won't have to test everything that's not an object because you can assume it'll do it for you.

Reply:
The lodash implementation pulled in a fairly large set other deps last time I checked it.

Reply:
Sure but we're already pulling them in everywhere this is used so should not add to the overall bundle size.

Reply:
I seem to remember it also doesn't cover as many cases as this implementation, but let me do a test run and report back for your decision.